### PR TITLE
bazel: Replace git_repository occurence by http_archive

### PR DESCRIPTION
**Pull request by jtattermusch**
Fixes https://github.com/grpc/grpc/issues/20524

### CODE REVIEW

File: bazel/grpc_python_deps.bzl [hunk 1]
```diff
@@ -41,13 +40,12 @@ def grpc_python_deps():
         )
 
     if "io_bazel_rules_python" not in native.existing_rules():
-        git_repository(
-            name = "io_bazel_rules_python",
-            commit = "fdbb17a4118a1728d19e638a5291b4c4266ea5b8",
-            remote = "https://github.com/bazelbuild/rules_python.git",
+        http_archive(
+            name = "rules_python",
```
Comment (ejona86) [inline bazel/grpc_python_deps.bzl]: You need to keep the old name if you aren't going to rename the usages.

File: bazel/grpc_python_deps.bzl [hunk 2]
```diff
@@ -41,13 +40,12 @@ def grpc_python_deps():
         )
 
     if "io_bazel_rules_python" not in native.existing_rules():
```
Comment (ejona86) [inline bazel/grpc_python_deps.bzl]: If you change the name, this would need to be changed, too.

File: bazel/grpc_python_deps.bzl (hunk 1 above)
Comment (jtattermusch) [reply bazel/grpc_python_deps.bzl]: Good catch, renamed.

**Comment by jtattermusch** [comment]
Tests looking better now.

**Comment by jtattermusch** [comment]
Known failures:
https://github.com/grpc/grpc/issues/20519

**Comment by jtattermusch** [comment]
@gnossen @veblush  need owners approval.

**Pull request merged**